compiler/rustc_const_eval/src/transform/validate.rs

```diff
@@ -564,13 +565,29 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
                     CastKind::PointerFromExposedAddress
                     | CastKind::PointerExposeAddress
                     | CastKind::Pointer(_) => {}
```

Comment:
There are still some variants here that have no checks? Should have a FIXME at least. Also the comment above seems outdated.

Reply:
They were not part of the previous PR so I didn't touch em